src/Tribe/Views/V2/Widgets/Widget_List.php

```diff
@@ -65,41 +73,122 @@ class Widget_List extends Widget_Abstract {
+		$updated_instance['title']                = wp_strip_all_tags( $new_instance['title'] );
+		$updated_instance['limit']                = $new_instance['limit'];
+		$updated_instance['no_upcoming_events']   = ! empty( $new_instance['no_upcoming_events'] );
+		$updated_instance['featured_events_only'] = ! empty( $new_instance['featured_events_only'] );
+		$updated_instance['jsonld_enable']        = (int) ( ! empty( $new_instance['jsonld_enable'] ) );
+		$updated_instance['tribe_is_list_widget'] = ! empty( $new_instance['tribe_is_list_widget'] );
+
+		return $this->filter_updated_instance( $updated_instance, $new_instance );
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
-	public function get_arguments() {
-		$arguments = $this->arguments;
+	public function setup_admin_fields() {
 
-		$arguments['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
-		// @todo update name once this widget is ready to replace the existing list widget.
-		$arguments['name']                          = esc_html__( 'Events List V2', 'the-events-calendar' );
-		$arguments['widget_options']['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
+		return [
+			'title'                => [
+				'label' => _x( 'Title:', 'The label for the field of the title of the List Widget.', 'the-events-calendar' ),
+				'type'  => 'text',
+			],
+			'limit'                => [
+				'label'   => _x( 'Show:', 'The label for the amount of events to show in the List Widget.', 'the-events-calendar' ),
+				'type'    => 'dropdown',
+				'options' => $this->get_limit_options(),
+			],
+			'no_upcoming_events'   => [
+				'label' => _x( 'Show widget only if there are upcoming events', 'The label for the option to hide the List Widget if no upcoming events.', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+			'featured_events_only' => [
+				'label' => _x( 'Limit to featured events only', 'The label for the option to only show featured events in the List Widget', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+			'jsonld_enable'        => [
+				'label' => _x( 'Generate JSON-LD data', 'The label for the option to enable JSONLD in the List Widget.', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+
+		];
+	}
+
+	/**
+	 * Get the options to use in a the limit dropdown.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> An array of options with the text and value included.
+	 */
+	public function get_limit_options() {
+		/**
+		 * Filter the max limit of events to display in the List Widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param int The max limit of events to display in the List Widget, default 10.
+		 */
+		$events_limit = apply_filters( 'tribe_events_widget_list_events_max_limit', 10 );
+
+		$options = [];
+
+		foreach ( range( 1, $events_limit ) as $i ) {
+			$options[] = [
+				'text'  => $i,
+				'value' => $i,
+			];
+		}
+
+		return $options;
+	}
+
+	/**
```

Comment:
* :no_entry_sign: Doc comment for parameter "$arguments" missing
* :no_entry_sign: Doc comment for parameter "$context" missing

<sub>hash: c898abc2ab174f8e3adde3d668077748</sub>